src/vm/i386/cgenx86.cpp

```diff
@@ -299,7 +299,12 @@ void TransitionFrame::UpdateRegDisplay(const PREGDISPLAY pRD)
 
     MethodDesc * pFunc = GetFunction();
     _ASSERTE(pFunc != NULL);
+
+#ifdef WIN64EXCEPTIONS
+    UpdateRegDisplayHelper(pRD, 0);
```

Comment:
@janvorli I'm sorry for unclear comment.

Suppose that a managed method ``A`` calls a  native method, and this native method throws an exception that ``A`` will handle.

``FindNonvolatileRegisterPointers`` attempts to find the frame of ``A`` using ``uOriginalSP`` which comes from ``Thread::VirtualUnwindToFirstManagedCallFrame``.

Note that the last step of ``VirtualUnwindToFirstManagedCallFrame`` will be ``PAL_VirtualUnwind`` (instead of ``UnwindStackFrame``) which does not consider callee-poped stack elements, and thus  ``uOriginalSP`` will be exactly same as Caller-SP of the native method.

(As I understand) stack walker will get the stack frame of ``A`` via calling `TransitionFrame::UpdateRegDisplay``, and the SP of ``A`` will be native method's Caller-SP + the result of CbStackPop() in the previous form.

Due to this mismatch, stack walker failed to recognize ``A``'s frame, which results in an assertion failure inside ``FindNonvolatileRegisterPointers``.

Please let me know if there is something wrong. I find all of these steps via execution log, but there may be something wrong.